src/dotnet/commands/dotnet-install/dotnet-install-tool/ProjectRestorer.cs

```diff
@@ -48,6 +48,7 @@ public ProjectRestorer(IReporter reporter = null)
             {
                 "--runtime",
                 AnyRid,
+                "--ignore-failed-sources", //nuget by default will throw when a feed cannot be reached
```

Comment:
This isn't the clarification I wanted (though adding nuget is fine too). What I wanted to see written down was it will throw when referencing a wildcard `*` version, which we do for tools, right? Without the wildcard, it can find the version on any feed so missing feeds don't matter if all packages are found. With the wildcard it only knows if it found the highest version if it hit all feeds. 

That's my understanding anyway. Do I have it right? If so, it's worth noting here because it's very subtle.

Reply:
This change has nothing to do with wildcard. If there is a broken feed in your nuget.config, regardless of what you are looking for, it will throw.
```
 <packageSources>
    <add key="nuget.org" value="https://api.nuget.org/v3/index.json/oooooops" protocolVersion="3" /> <-- it will throw due to this
    <add key="dotnet-core" value="https://dotnet.myget.org/F/dotnet-core/api/v3/index.json" />
  </packageSources>
```

However the more I think about it, we should has an opinion on `dotnet install` just like `dotnet restore`. Implicitly adding this flag can cause non deterministic

Reply:
> nothing to do with wildcard

Are you sure? Try this:

nuget.config
``` xml
<configuration>
  <packageSources>
   <clear />
   <add key="bogus" value="https://example.com/not_a_real_feed/index.json" />
   <add key="nuget" value="https://api.nuget.org/v3/index.json" />
  </packageSources>
</configuration>
```

``` xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp2.0</TargetFramework>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="NewtonSoft.Json" Version="10.0.1" />
  </ItemGroup>
</Project>
```

Restore with specific version in PackageRef, works:
````
> dotnet restore
  Restore completed in 39.7 ms for D:\Temp\aaaa\aaaa.csproj.
```

Update PackageRef to use wildcard:
```
    <PackageReference Include="NewtonSoft.Json" Version="10.0.*" />
```

Restore with wild card, fails:
```
> dotnet restore
  Restoring packages for D:\Temp\aaaa\aaaa.csproj...
C:\Program Files\dotnet\sdk\2.1.100\NuGet.targets(104,5): error : Unable to load the service index for source https://example.com/not_a_real_feed/index.json. [D:\Temp\aaaa\aaaa.csproj]
C:\Program Files\dotnet\sdk\2.1.100\NuGet.targets(104,5): error :   Response status code does not indicate success: 404 (Not Found). [D:\Temp\aaaa\aaaa.csproj]
```

If I ask for a specific version, then it can get it from any feed, but if I ask for * it has to ask all the feeds to determine the highest version to use. If all the versions I request are explicit, then there is no non-determinism when a feed is down. If I find all the packages, it doesn't matter that some of my feeds are down. 

My point since the beginning here is that I've certainly ran with feeds that are down before, but I never ever saw the error that --ignore-failed-sources suppresses. This is how I guessed that the wildcard is at play and the experiment above seems to confirm my suspicion. 

Am I missing something?

Reply:
I think I miss something this time. It does have something to do with wildcard. 

However now it is more close to `dotnet add package`, and i find an existing inconsistency. 

If you have a bogus feed and do a  `dotnet add package Newtonsoft.Json`. I will fail, and `dotnet add package` does not have ` --ignore-failed-sources`. So the user will have the similar failure in offline scenario today. 

And if we want to be consistent with dotnet add package, we do nothing?? Let me dig more, and see if i can repo offline dotnet add package  
```
λ  dotnet add package Newtonsoft.Json
  Writing C:\Users\wul\AppData\Local\Temp\tmpC058.tmp
info : Adding PackageReference for package 'Newtonsoft.Json' into project 'C:\Users\wul\Downloads\console5\console5.csproj'.
log  : Restoring packages for C:\Users\wul\Downloads\console5\console5.csproj...
info :   GET https://dotnet.myget.org/F/dotnet-core/api/v3/index.jsonssssssssssssss/FindPackagesById()?id='Newtonsoft.Json'
info :   GET https://api.nuget.org/v3-flatcontainer/newtonsoft.json/index.json
info :   OK https://api.nuget.org/v3-flatcontainer/newtonsoft.json/index.json 233ms
info :   NotFound https://dotnet.myget.org/F/dotnet-core/api/v3/index.jsonssssssssssssss/FindPackagesById()?id='Newtonsoft.Json' 2237ms
log  : Retrying 'FindPackagesByIdAsyncCore' for source 'https://dotnet.myget.org/F/dotnet-core/api/v3/index.jsonssssssssssssss/FindPackagesById()?id='Newtonsoft.Json''.
log  : Response status code does not indicate success: 404 (Not Found).
info :   GET https://dotnet.myget.org/F/dotnet-core/api/v3/index.jsonssssssssssssss/FindPackagesById()?id='Newtonsoft.Json'
info :   GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/newtonsoft.json/index.json
info :   OK https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/newtonsoft.json/index.json 133ms
info :   GET https://www.myget.org/F/wultest/api/v3/flatcontainer/newtonsoft.json/index.json
info :   NotFound https://www.myget.org/F/wultest/api/v3/flatcontainer/newtonsoft.json/index.json 275ms
info :   NotFound https://dotnet.myget.org/F/dotnet-core/api/v3/index.jsonssssssssssssss/FindPackagesById()?id='Newtonsoft.Json' 5545ms
log  : Retrying 'FindPackagesByIdAsyncCore' for source 'https://dotnet.myget.org/F/dotnet-core/api/v3/index.jsonssssssssssssss/FindPackagesById()?id='Newtonsoft.Json''.
log  : Response status code does not indicate success: 404 (Not Found).
info :   GET https://dotnet.myget.org/F/dotnet-core/api/v3/index.jsonssssssssssssss/FindPackagesById()?id='Newtonsoft.Json'
info :   NotFound https://dotnet.myget.org/F/dotnet-core/api/v3/index.jsonssssssssssssss/FindPackagesById()?id='Newtonsoft.Json' 7083ms
error: Failed to retrieve information about 'Newtonsoft.Json' from remote source 'https://dotnet.myget.org/F/dotnet-core/api/v3/index.jsonssssssssssssss/FindPackagesById()?id='Newtonsoft.Json''.
error:   Response status code does not indicate success: 404 (Not Found).
```

Reply:
I get repo on offline `dotnet add`. Note, you need to clean nuget cache first. 

```
dotnet add package microsoft.netcore.platforms
  Writing /var/folders/_0/_t4443kj1nd02pjqqzl79h2h0000gn/T/tmpC2jKpA.tmp
info : Adding PackageReference for package 'microsoft.netcore.platforms' into project '/Users/williamli/Downloads/console2/console2.csproj'.
log  : Restoring packages for /Users/williamli/Downloads/console2/console2.csproj...
error: Unable to load the service index for source https://api.nuget.org/v3/index.json.
error:   An error occurred while sending the request.
error:   Couldn't resolve host name
```

So to keep the experience consistent with `dotnet add` **we should not fix it**? More, offline experience is needed due to `dotnet-watch`. But after preview 1, dotnet-watch is bundled, there is no dotnet install involved. And there is no tool packages lives in offline cache in preview2 time.

@KathleenDollard @livarcocc @peterhuene @nguerrera

Reply:
I think dotnet add and dotnet install should behave consistently, but it's not clear to me that we can't decide to change dotnet add.  This definitely needs more discussion.

Reply:
If we want to fix it, do we want implicit and always ignore feed failure or explicit flag passing in.

Reply:
at least dotnet add is hard to change in technical sense. dotnet add will pass through to `xplat.nuget package add`. the code is in nuget